src/main/java/ru/sbtqa/tag/pagefactory/PageFactory.java

```diff
@@ -172,4 +173,18 @@ public static Environment getEnvironment() {
                 }
         }
     }
+
+    /**
+     * @return the IS_SHARING_PROCESSING
+     */
+    public static boolean isSharingProcessing() {
+        return isSharingProcessing;
+    }
+
+    /**
+     * @param isSharingProcessing the IS_SHARING_PROCESSING to set
+     */
+    public static void setSharingProcessing(boolean isSharingProcessing) {
+        isSharingProcessing = isSharingProcessing;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove or correct this useless self-assignment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1656)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "isSharingProcessing". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "isSharingProcessing". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)